thunder/core/update_aliases.py

```diff
@@ -0,0 +1,106 @@
+from functools import reduce, partial
+
+import thunder.core.prims as prims
+from thunder.core.proxies import TensorProxy, variableify, unvariableify
+from thunder.core.trace import from_trace, tracectx, TraceCtx as Trace, TraceProvenance
+
+
+def _get_update_bsym(group, swap_map, new_aliases):
+    aliases = tuple(unvariableify(alias) for alias in group)
+    update_aliases_bsym = prims.update_aliases.bind(aliases, output=new_aliases)
+    update_aliases_bsym = update_aliases_bsym.from_bsym_swap_proxies(swap_map)
+    for new_alias, old_alias in zip(new_aliases, group):
+        swap_map[old_alias] = new_alias
+    return update_aliases_bsym, swap_map
+
+
+def _get_new_alias(alias, trace):
+    with tracectx(trace):
+        new_alias = TensorProxy(like=alias.proxy, requires_grad=alias.proxy.requires_grad)
+    return new_alias
+
+
+def _get_new_aliases(aliases, trace):
+    return tuple(map(partial(_get_new_alias, trace=trace), aliases))
+
+
+def _is_inplace_op(bsym):
+    return (bsym.sym.tags and prims.OpTags.IN_PLACE in bsym.sym.tags) or (
+        bsym.subsymbols and bsym.subsymbols[-1].sym.id == prims.PrimIDs.COPY_
+    )
+
+
+def _is_view_creation_op(bsym):
+    import thunder.torch as ltorch
+
+    return bsym.sym in ltorch._syms_returning_views or bsym.sym in ltorch._syms_that_may_return_views
```

Comment:
Would we happen to have corner cases to check behavior/outputs of `insert_alias_updates` when `_syms_that_may_return_views` ops return a new tensor not a view?

Reply:
Perhaps eventually.  The worry for calling `update_aliases` too frequently would be that bsyms that could have otherwise been reordered no longer can, potentially resulting in slower performance.  The added complexity of dealing with these corner cases needs to be weighed against this as of yet unknown performance cost.